In [1]:
import numpy as np
import pandas as pd
import re
from PyPDF2 import PdfFileReader

# Load the PDF File

In [2]:
# open the PDF file
pdfFile = open('/Users/johnathonsmith/Downloads/293RDCV.PEND.CASE_1642601189564_-1416322429.pdf', 'rb')


# Read the File

In [3]:
# create PDFFileReader object to read the file
pdfReader = PdfFileReader(pdfFile)

# Check For Encryption

In [4]:
if pdfReader.isEncrypted == True:
    print("PDF File is encrypted. Password is needed to continue.")
else:
    print("PDF File is not encrypted. Password is not needed.")

PDF File is encrypted. Password is needed to continue.


# Print Document Info

In [5]:
#print("PDF File name: " + str(pdfReader.getDocumentInfo().title))
#print("PDF File created by: " + str(pdfReader.getDocumentInfo().creator))
#print("- - - - - - - - - - - - - - - - - - - -")

# Try Using PDFminer3

In [6]:
#imports
import pandas as pd

from pdfminer3.layout import LAParams, LTTextBox
from pdfminer3.pdfpage import PDFPage
from pdfminer3.pdfinterp import PDFResourceManager
from pdfminer3.pdfinterp import PDFPageInterpreter
from pdfminer3.converter import PDFPageAggregator
from pdfminer3.converter import TextConverter
import io

In [7]:
#Set up resource manager to handle pdf content. text, images, etc.
resource_manager = PDFResourceManager()

#Used to display text
fake_file_handle = io.StringIO()

#Set up converter
converter = TextConverter(resource_manager, fake_file_handle, laparams=LAParams())

#Set up page interpreter
page_interpreter = PDFPageInterpreter(resource_manager, converter)

with open('/Users/johnathonsmith/Downloads/293RDCV.PEND.CASE_1642601189564_-1416322429.pdf', 'rb') as fh:

    for page_number, page in enumerate(PDFPage.get_pages(fh,
                                  caching=True,
                                  maxpages = 14,
                                  check_extractable=True)):
        #Only read page 1 for now (index = 0)
        if page_number == 0:
            page_interpreter.process_page(page)

    text = fake_file_handle.getvalue()

# close open handles
converter.close()
fake_file_handle.close()

print(text)

                COUNTY OF MAVERICK 293RD DISTRICT COURT CIVIL PENDING CASES AS OF 01/19/2022

                                           RAN ON JANUARY 19, 2022 09:02am PAGE: 1

        CAUSE #             FILE DATE CAUSE OF ACTION           DOCKT DATE  DOCKET TYPE       ANS FILED  CR CASE #

PLAINTIFF NAME                PLAINTIFF ATTORNEY        DEFENDANT NAME                DEFENDANT ATTORNEY

        7815-OTH           02/22/1985 1                                                                  

        ATTORNEY GENERAL              MARIA LARSON              MENDEZ, CARLOS GILBERTO                            
        MENDEZ, OLGA LIDIA                                                                                         
        -------------------------------------------------------------------------------------------------------------
        90-11-10256-CV     11/08/1990 1                                                                  

        ATTORNEY GENERAL            

In [8]:
#Now try finding places to break it down a bit
print(repr(text))

'                COUNTY OF MAVERICK 293RD DISTRICT COURT CIVIL PENDING CASES AS OF 01/19/2022\n\n                                           RAN ON JANUARY 19, 2022 09:02am PAGE: 1\n\n        CAUSE #             FILE DATE CAUSE OF ACTION           DOCKT DATE  DOCKET TYPE       ANS FILED  CR CASE #\n\nPLAINTIFF NAME                PLAINTIFF ATTORNEY        DEFENDANT NAME                DEFENDANT ATTORNEY\n\n        7815-OTH           02/22/1985 1                                                                  \n\n        ATTORNEY GENERAL              MARIA LARSON              MENDEZ, CARLOS GILBERTO                            \n        MENDEZ, OLGA LIDIA                                                                                         \n        -------------------------------------------------------------------------------------------------------------\n        90-11-10256-CV     11/08/1990 1                                                                  \n\n        ATTORNEY GEN

In [9]:
#First, strip leading and trailing whitespaces
text = text.strip()
print(repr(text))

'COUNTY OF MAVERICK 293RD DISTRICT COURT CIVIL PENDING CASES AS OF 01/19/2022\n\n                                           RAN ON JANUARY 19, 2022 09:02am PAGE: 1\n\n        CAUSE #             FILE DATE CAUSE OF ACTION           DOCKT DATE  DOCKET TYPE       ANS FILED  CR CASE #\n\nPLAINTIFF NAME                PLAINTIFF ATTORNEY        DEFENDANT NAME                DEFENDANT ATTORNEY\n\n        7815-OTH           02/22/1985 1                                                                  \n\n        ATTORNEY GENERAL              MARIA LARSON              MENDEZ, CARLOS GILBERTO                            \n        MENDEZ, OLGA LIDIA                                                                                         \n        -------------------------------------------------------------------------------------------------------------\n        90-11-10256-CV     11/08/1990 1                                                                  \n\n        ATTORNEY GENERAL            

In [10]:
#Now separate header and body text info

#For header
header = text[:392]
print(repr(header))

'COUNTY OF MAVERICK 293RD DISTRICT COURT CIVIL PENDING CASES AS OF 01/19/2022\n\n                                           RAN ON JANUARY 19, 2022 09:02am PAGE: 1\n\n        CAUSE #             FILE DATE CAUSE OF ACTION           DOCKT DATE  DOCKET TYPE       ANS FILED  CR CASE #\n\nPLAINTIFF NAME                PLAINTIFF ATTORNEY        DEFENDANT NAME                DEFENDANT ATTORNEY\n\n        '


In [11]:
#For body
text = text[392:]
print(repr(text))

'7815-OTH           02/22/1985 1                                                                  \n\n        ATTORNEY GENERAL              MARIA LARSON              MENDEZ, CARLOS GILBERTO                            \n        MENDEZ, OLGA LIDIA                                                                                         \n        -------------------------------------------------------------------------------------------------------------\n        90-11-10256-CV     11/08/1990 1                                                                  \n\n        ATTORNEY GENERAL              ROLANDO MENCHACA          MARTINEZ, HECTOR RENE                              \n        MARTINEZ, ROSALINDA                                                                                        \n        -------------------------------------------------------------------------------------------------------------\n        90-12-10326-CV     12/14/1990 INJURY OR DAMAGE OTHER T                     

In [12]:
#Now split each case by the dashed lines
cases = text.split('-------------------------------------------------------------------------------------------------------------')
cases

['7815-OTH           02/22/1985 1                                                                  \n\n        ATTORNEY GENERAL              MARIA LARSON              MENDEZ, CARLOS GILBERTO                            \n        MENDEZ, OLGA LIDIA                                                                                         \n        ',
 '\n        90-11-10256-CV     11/08/1990 1                                                                  \n\n        ATTORNEY GENERAL              ROLANDO MENCHACA          MARTINEZ, HECTOR RENE                              \n        MARTINEZ, ROSALINDA                                                                                        \n        ',
 '\n        90-12-10326-CV     12/14/1990 INJURY OR DAMAGE OTHER T                                           \n\n        DIAZ CARPENTER, ENRIQUETA     JAY T. KIMBROUGH          BOWLES JR., RUDY              RICHARD TINSMAN      \n        ',
 '\n        91-06-10598-CV     10/22/1993 MOTION ENFO

In [13]:
len(cases)

8

In [14]:
cases[0]

'7815-OTH           02/22/1985 1                                                                  \n\n        ATTORNEY GENERAL              MARIA LARSON              MENDEZ, CARLOS GILBERTO                            \n        MENDEZ, OLGA LIDIA                                                                                         \n        '

In [15]:
#Only work with one case for now
test_case = cases[1]
print(repr(test_case))

'\n        90-11-10256-CV     11/08/1990 1                                                                  \n\n        ATTORNEY GENERAL              ROLANDO MENCHACA          MARTINEZ, HECTOR RENE                              \n        MARTINEZ, ROSALINDA                                                                                        \n        '


In [16]:
#Strip the leading white space
test_case = test_case.lstrip()
print(repr(test_case))

'90-11-10256-CV     11/08/1990 1                                                                  \n\n        ATTORNEY GENERAL              ROLANDO MENCHACA          MARTINEZ, HECTOR RENE                              \n        MARTINEZ, ROSALINDA                                                                                        \n        '


In [17]:
#Try splitting on the new lines
case_info = test_case.split('\n')

In [18]:
len(case_info)

5

In [19]:
#I can isolate the cause number, file date, cause of action, docket date, docket type, 
#ans filed date, and cr number from here
case_info[0]

'90-11-10256-CV     11/08/1990 1                                                                  '

In [20]:
#Date starts 19 characters in.
case_info[0][19:]

'11/08/1990 1                                                                  '

In [21]:
#Cause of action start 30 characters in
case_info[0][30:]

'1                                                                  '

In [22]:
#Docket date starts 56 characters in
case_info[0][56:]

'                                         '

In [23]:
#Docket type starts 68 characters in
case_info[0][68:]

'                             '

In [24]:
#ANS Filed date starts 86 characters in
case_info[0][86:]

'           '

In [25]:
#CR number starts after 96 characters on the first line
case_info[0][96:]

' '

In [26]:
#Appears to be an extra new line
case_info[1]

''

In [27]:
#The first line of names starts 8 characters in. This should be the same for each case
case_info[2][8:]

'ATTORNEY GENERAL              ROLANDO MENCHACA          MARTINEZ, HECTOR RENE                              '

__Name Spacing__

The names appear to all start at the same place every time. Here is what I have found:

* Plaintiff Name: 8 Characters in
* Plaintiff Attorney: 38 Characters in
* Defendant Name: 64 Characters in
* Defendant Attorney: 94 Characters in

These numbers seem consistent regardless of what line they appear on.

In [28]:
#The second name starts 38 characters in. I hope this is the same for all of them
case_info[2][38:]

'ROLANDO MENCHACA          MARTINEZ, HECTOR RENE                              '

In [29]:
#The third name starts 64 characters in. I hope this is also the same for all of them
case_info[2][64:]

'MARTINEZ, HECTOR RENE                              '

In [30]:
#The fourth name starts 94 characters in
case_info[2][94:]

'                     '

In [31]:
case_info[3]

'        MARTINEZ, ROSALINDA                                                                                        '

In [32]:
#The second plaintiff name starts 8 characters in. This should be the same for each case
case_info[3][8:]

'MARTINEZ, ROSALINDA                                                                                        '

In [33]:
#The second defendant name starts 64 characters in.
case_info[3][64:]

'                                                   '

# Start Isolating Info

Isolate the case info and add it to a temp dict. When complete the temp dict will be appended to the container dictionary. Finally, when the info for each case has been extracted, convert the container dict to a dataframe.

In [34]:
#Create a container dict to store all case info
container_dict = {}

#Create a temp dict that stores individual case info
case_dict = {}

__County__

The county name will be found in the header info of each page. The easiest way I can think to isolate the county is to just use an if statement and check for each county name in the header text. There should only be one match, so it should be fine. Possible counties: Maverick, Dimmit, Zavala

In [35]:
header

'COUNTY OF MAVERICK 293RD DISTRICT COURT CIVIL PENDING CASES AS OF 01/19/2022\n\n                                           RAN ON JANUARY 19, 2022 09:02am PAGE: 1\n\n        CAUSE #             FILE DATE CAUSE OF ACTION           DOCKT DATE  DOCKET TYPE       ANS FILED  CR CASE #\n\nPLAINTIFF NAME                PLAINTIFF ATTORNEY        DEFENDANT NAME                DEFENDANT ATTORNEY\n\n        '

In [36]:
#Use if statement to check for county names inside the header info
if header.count('MAVERICK') >= 1:
    county = 'Maverick'
elif header.count('DIMMIT') >= 1:
    county = 'Dimmit'
elif header.count('ZAVALA') >= 1:
    county = 'Zavala'
else:
    county = 'Something went wrong!'
    
county

'Maverick'

In [37]:
#Add to dict
case_dict['County'] = county

__Cause Number__

In [38]:
#Get the cause number
cause_num = case_info[0][:19].strip()
cause_num

'90-11-10256-CV'

In [39]:
#Add to dict
case_dict['Cause Number'] = cause_num

__File Date__

In [40]:
file_date = case_info[0][15:30].strip()
file_date

'11/08/1990'

In [41]:
#Add to dict
case_dict['File Date'] = file_date

__Cause of Action__

In [42]:
coa = case_info[0][30:56].strip()
coa

'1'

In [43]:
#Add to dict
case_dict['Cause of Action'] = coa

__Docket Date__

In [44]:
docket_date = case_info[0][56:68].strip()
docket_date

''

In [45]:
#Add to dict
case_dict['Docket Date'] = docket_date

__Docket Type__

In [46]:
docket_type = case_info[0][68:86].strip()
docket_type

''

In [47]:
#Add to dict
case_dict['Docket Type'] = docket_type

__ANS File Date__

In [48]:
ans_date = case_info[0][86:].strip()
ans_date

''

In [49]:
#Add to dict
case_dict['ANS Filed'] = ans_date

__CR Number__

In [50]:
cr_num = case_info[0][96:].strip()
cr_num

''

In [51]:
#Add to dict
case_dict['CR Number'] = cr_num

__Names__

The next few entries will be tricky. If there is more than one name associated with each label (plaintiff name, plaintiff attorney, defendant name, defendant attorney), then there will be more than one line of names. In my function, I'll have to set up a loop that can accurately pick up all of the names. For now, I'll make each label a list of names. If it needs to be changed to a string of names later, I can do that too. I'll also have to build a function that checks to see if each segment is composed entirely of whitespace. That way, if it is, we know there isn't a name for that particular label. 

In [52]:
#Will need to set up the lists before the loop. They will be added to the dict after the loop
plaintiff_names = []
plaintiff_attorneys = []
defendant_names = []
defendant_attorneys = []

#The names always start on the third line (index = 2)
for i in range(2, len(case_info)):
    #Get the plaintiff name on current line
    plaintiff_name = case_info[i][:38]
    
    #Check if plaintiff_name is all whitesapace. If not, strip it and add to names list
    #Also check that the string is not empty
    if plaintiff_name.isspace() == False and len(plaintiff_name) > 0:
        plaintiff_names.append(plaintiff_name.strip())
        
    #Get the plaintiff attorney on current line
    plaintiff_attorney = case_info[i][38:64]
    
    #Check if plaintiff_attorney is all whitespace. If not, strip it and add to list
    #Also check that the string is not empty
    if plaintiff_attorney.isspace() == False and len(plaintiff_attorney) > 0:
        plaintiff_attorneys.append(plaintiff_attorney.strip())
        
    #Get the defendant name on current line
    defendant_name = case_info[i][64:94]
    
    #Check if defendant_name is all whitespace. If not, strip it and add to list
    #Also check that the string is not empty
    if defendant_name.isspace() == False and len(defendant_name) > 0:
        defendant_names.append(defendant_name.strip())
    
    #Get the defendant attorney on current line
    defendant_attorney = case_info[i][94:]
    
    #Check if the defendant_attorney is all whitespace. If not, strip it and add to list
    #Also check that the string is not empty
    if defendant_attorney.isspace() == False and len(defendant_attorney) > 0:
        defendant_attorneys.append(defendant_attorney.strip())

In [53]:
#Now add all names to the dict
case_dict['Plaintiff Name'] = plaintiff_names
case_dict['Plaintiff Attorney'] = plaintiff_attorneys
case_dict['Defendant Name'] = defendant_names
case_dict['Defendant Attorney'] = defendant_attorneys

In [54]:
case_dict

{'County': 'Maverick',
 'Cause Number': '90-11-10256-CV',
 'File Date': '11/08/1990',
 'Cause of Action': '1',
 'Docket Date': '',
 'Docket Type': '',
 'ANS Filed': '',
 'CR Number': '',
 'Plaintiff Name': ['ATTORNEY GENERAL', 'MARTINEZ, ROSALINDA'],
 'Plaintiff Attorney': ['ROLANDO MENCHACA'],
 'Defendant Name': ['MARTINEZ, HECTOR RENE'],
 'Defendant Attorney': []}

# Automate Process

Now that I can gather all available info for any single case on a page, build a function that can retrieve it all for all cases on all pages. Then convert the data to a dataframe and return it.

In [55]:
def extract_pdf_data(text):
    """
    This function takes in a page of text from a pdf file. It will extract all available information for each case on
    the page, put it all into dictionaries, and then return a list containing the other dictionaries for each
    case.
    
    Parameter:
        - text: The text of a single pdf page.
        
    Returns:
        - cases: A list containing the dictionaries of all cases.
    """
    
    #Set up the container list
    case_dicts = []
    
    #First, strip leading and trailing whitespaces
    text = text.strip()
    
    #Separate header and body text data
    #For header
    header = text[:389]
    
    #For body
    text = text[392:]
    
    #Remove headers from all subsequent pages
    text = re.sub(r"""COUNTY OF .{6,8} 293RD DISTRICT COURT CIVIL PENDING CASES AS OF .{10}

                                           RAN ON .{20,26} PAGE: \d{1,3}

        CAUSE #             FILE DATE CAUSE OF ACTION           DOCKT DATE  DOCKET TYPE       ANS FILED  CR CASE #

PLAINTIFF NAME                PLAINTIFF ATTORNEY        DEFENDANT NAME                DEFENDANT ATTORNEY""", '', text)
    
    #Use if statement to check for county names inside the header info
    if header.count('MAVERICK') >= 1:
        county = 'Maverick'
    elif header.count('DIMMIT') >= 1:
        county = 'Dimmit'
    elif header.count('ZAVALA') >= 1:
        county = 'Zavala'
    else:
        county = 'Something went wrong!'
    
    #Now split each case by the dashed lines
    cases = text.split('-------------------------------------------------------------------------------------------------------------')
    
    #Drop the last case. It only consists of the count of cases in the pdf. Not needed here
    cases.pop()
    
    #Loop through each case and build a dictionary with its info
    for case in cases:
        
        #Initialize the temp_dict
        temp_dict = {}
        
        #Remove leading whitespace only
        case = case.lstrip()
        
        #Split on the '/n'
        case_info = case.split('\n')
        
        #Gather the case data
        case_list = collect_case_data(case_info)
        
        #Finally, append this dictionary to the container dict
        case_dicts.extend(case_list)
        
    return case_dicts
        

In [56]:
#Create a function the collects data from combined cases
def collect_case_data(case_info):
    """
    This function takes in a list of strings that represent a single case. Each string may contain relevant information
    that will need to be collected, and added to a dictionary. That dictionary will then be added to a list 
    and returned.
    
    WARNING: This function is recursive.
    
    Parameter:
        -case_info: The list of strings representing a single case
        
    Returns:
        -case_list: A list containing the case dictionaries
    """
    
    #Initialize the temp list to store the dict(s)
    case_list = []
    
    #Initialize the temp dict
    temp_dict = {}
    
    #Remove leading whitespace
    case_info[0] = case_info[0].lstrip()

    #Get the cause number
    cause_num = case_info[0][:19].strip()

    #Get the file date
    file_date = case_info[0][15:30].strip()

    #Get the Cause of Action
    coa = case_info[0][30:56].strip()

    #Get the docket_date
    docket_date = case_info[0][56:68].strip()

    #Get the docket type
    docket_type = case_info[0][68:86].strip()

    #Get the ANS File
    ans_date = case_info[0][86:].strip()

    #Get the CR Number
    cr_num = case_info[0][96:].strip()

    #Get the names associated with each label
    #Will need to set up the lists before the loop. They will be added to the dict after the loop
    plaintiff_names = []
    plaintiff_attorneys = []
    defendant_names = []
    defendant_attorneys = []

    #The names always start on the third line (index = 2)
    for i in range(2, len(case_info)):

        #Check that this line is not the start of a combined case. If it is, collect combined case data,
        #Then break from this loop
        if check_for_combined_case(case_info[i][:38]) == True:
            #Set up temporary dictionary
            combined_case = {}

            #Find combined case text
            combined_case_text = []

            #Loop through and grab the necessary lines from the current case
            for index in range(i, len(case_info)):
                combined_case_text.append(case_info[index])

            #Collect combined case data
            combined_case = collect_case_data(combined_case_text)

            #Add the temp dictionary to the overall container dict
            case_list.extend(combined_case)

            #Finally, break out of the loop
            break


        #Get the plaintiff name on current line
        plaintiff_name = case_info[i][:38]

        #Check if plaintiff_name is all whitesapace. If not, strip it and add to names list
        #Also check that the string is not empty
        if plaintiff_name.isspace() == False and len(plaintiff_name) > 0:
            plaintiff_names.append(plaintiff_name.strip())

        #Get the plaintiff attorney on current line
        plaintiff_attorney = case_info[i][38:64]

        #Check if plaintiff_attorney is all whitespace. If not, strip it and add to list
        #Also check that the string is not empty
        if plaintiff_attorney.isspace() == False and len(plaintiff_attorney) > 0:
            plaintiff_attorneys.append(plaintiff_attorney.strip())

        #Get the defendant name on current line
        defendant_name = case_info[i][64:94]

        #Check if defendant_name is all whitespace. If not, strip it and add to list
        #Also check that the string is not empty
        if defendant_name.isspace() == False and len(defendant_name) > 0:
            defendant_names.append(defendant_name.strip())

        #Get the defendant attorney on current line
        defendant_attorney = case_info[i][94:]

        #Check if the defendant_attorney is all whitespace. If not, strip it and add to list
        #Also check that the string is not empty
        if defendant_attorney.isspace() == False and len(defendant_attorney) > 0:
            defendant_attorneys.append(defendant_attorney.strip())


    #Now put all the info into a temp dict. Check that each entry actually exists though
    #If the entry does not exist, just enter None
    temp_dict['County'] = county
    temp_dict['Cause Number'] = cause_num
    temp_dict['File Date'] = file_date
    temp_dict['Cause of Action'] = coa
    temp_dict['Docket Date'] = docket_date
    temp_dict['Docket Type'] = docket_type
    temp_dict['ANS File'] = ans_date
    temp_dict['CR Number'] = cr_num
    temp_dict['Plaintiff Name'] = plaintiff_names
    temp_dict['Plaintiff Attorney'] = plaintiff_attorneys
    temp_dict['Defendant Name'] = defendant_names
    temp_dict['Defendant Attorney'] = defendant_attorneys
    
    #Append the temp_dict to the case_list
    case_list.append(temp_dict)
    
    return case_list
    

# Notes

* 12-02-27168-MCV: Has two dates under the file date. Do we need them both?
* The case following 15-11-32356-MCV does not have a standard cause number. It just says 293rd
* 18-02-35252-MCV: Was split into two different cases. Can be found on pages 43 and 44
* 19-09-37842-MCV: Was split into two different cases. Can be found on pages 67 and 68
* 19-09-37846-MCV: Was split into three different cases. Can be found on pages 69, 70, 71
* 19-09-37846-MCV-A: Was split into two different cases. Can be found on pages 86, 87
* 20-08-39164-MCV: Was split into two different cases. Can be found on pages 93, 94
* 20-09-05924-TX: Was split into two different cases. Can be found on pages 97, 98
* 20-12-39442-MCV: Was split into two different cases. Can be found on pages 103, 104
* 21-02-39635-MCV: Was split into two different cases. Can be found on pages 110, 111
* 21-03-39756-MCV: Was split into two different cases. Can be found on pages 115, 116
* 21-09-40432-MCV: Was split into two different cases. Can be found on pages 146, 147
* 21-10-40462-MCV: Was split into two different cases. Can be found on pages 148, 149
* 21-11-40584-MCV: Was split into two different cases. Can be found on pages 156, 157

In [61]:
def build_cases_dataframe(pdf_path):
    """
    This function reads in a pdf and extracts all available information for each case. It then returns a dataframe.
    
    Parameter:
        - pdf_path: The file path for the pdf to be read.
        
    Returns:
        - df: A dataframe of the resulting case information
    """
    
    #Set up the container list
    container_list = []
    
    #Set up resource manager to handle pdf content. text, images, etc.
    resource_manager = PDFResourceManager()

    #Used to display text
    fake_file_handle = io.StringIO()

    #Set up converter
    converter = TextConverter(resource_manager, fake_file_handle, laparams=LAParams())

    #Set up page interpreter
    page_interpreter = PDFPageInterpreter(resource_manager, converter)

    with open(pdf_path, 'rb') as fh:

        for page in PDFPage.get_pages(fh, caching=True, check_extractable=True):
            #Process the current page
            page_interpreter.process_page(page)
            
        #Save the current page's text to a variable
        text = fake_file_handle.getvalue()

        #Extract info from current page
        case_dicts = extract_pdf_data(text)

        #Extend case_dicts to container_dict
        container_list.extend(case_dicts)

    # close open handles
    converter.close()
    fake_file_handle.close()
    
    #For testing
    print(f"Collected Data From {len(container_list)} Cases.")
    
    #Build dataframe
    df = pd.DataFrame(container_list)
    
    return df

In [62]:
#For testing
df = build_cases_dataframe('/Users/johnathonsmith/Downloads/293RDCV.PEND.CASE_1642601189564_-1416322429.pdf')

Collected Data From 1179 Cases.


In [63]:
df

County     Cause Number   File Date           Cause of Action  \
0     Maverick         7815-OTH  02/22/1985                         1   
1     Maverick   90-11-10256-CV  11/08/1990                         1   
2     Maverick   90-12-10326-CV  12/14/1990  INJURY OR DAMAGE OTHER T   
3     Maverick   91-06-10598-CV  10/22/1993  MOTION ENFORCEMENT, TO M   
4     Maverick   95-01-12880-CV  01/05/1995                         1   
...        ...              ...         ...                       ...   
1174  Maverick  22-01-40738-MCV  01/14/2022        DIVORCE - CHILDREN   
1175  Maverick  22-01-40740-MCV  01/14/2022  DEBT/CONTRACT - DEBT COL   
1176  Maverick  22-01-40742-MCV  01/14/2022  INJURY OR DAMAGE INVOLVI   
1177  Maverick  22-01-40744-MCV  01/17/2022  DEBT/CONTRACT - DEBT COL   
1178  Maverick  22-01-40748-MCV  01/18/2022             JUDGMENT NISI   

     Docket Date       Docket Type ANS File CR Number  \
0                                                       
1                                                       
2                                                       
3     09/13/1994  MOTION TO ENFORC                      
4                                                       
...          ...               ...      ...       ...   
1174                                                    
1175                                                    
1176                                                    
1177                                                    
1178                                                    

                                  Plaintiff Name  \
0         [ATTORNEY GENERAL, MENDEZ, OLGA LIDIA]   
1        [ATTORNEY GENERAL, MARTINEZ, ROSALINDA]   
2                    [DIAZ CARPENTER, ENRIQUETA]   
3     [ATTORNEY GENERAL, FUENTES, MARIA EUGENIA]   
4      [ATTORNEY GENERAL, LARA, MARIA GUADALUPE]   
...                                          ...   
1174                           [ABOYTES CYNTHIA]   
1175                            [SYNCHRONY BANK]   
1176        [CASARES JR CARLOS, CASARES EDUARDO]   
1177               [CROWN ASSET MANAGEMENT, LLC]   
1178                             [SERNA ROBERTO]   

                      Plaintiff Attorney             Defendant Name  \
0                         [MARIA LARSON]  [MENDEZ, CARLOS GILBERTO]   
1                     [ROLANDO MENCHACA]    [MARTINEZ, HECTOR RENE]   
2                     [JAY T. KIMBROUGH]         [BOWLES JR., RUDY]   
3                     [ROLANDO MENCHACA]     [ARROYO, JUAN ANTONIO]   
4                    [ENRIQUE FERNANDEZ]             [LARA, HOMERO]   
...                                  ...                        ...   
1174                      [JAD P HARPER]           [ABOYTES ANDRES]   
1175                     [ZHANG JESSICA]             [CASAS EMILIO]   
1176  [VICTOR E. PERRY, VICTOR E. PERRY]            [FUENTES DAVID]   
1177              [BROGAN COLIN PATRICK]         [BALDERAS PAULA J]   
1178                                  []   [MEDIDORO ROILAN MARTIN]   

     Defendant Attorney  
0                    []  
1                    []  
2     [RICHARD TINSMAN]  
3                    []  
4                    []  
...                 ...  
1174                 []  
1175                 []  
1176                 []  
1177                 []  
1178                 []  

[1179 rows x 12 columns]

# Something Went Wrong

Just discovered that the PDF pages are not independent. As in, some case information can bleed over into the next page. So the process I used to extract the information won't work as is. I'll work on it some more tomorrow.

__Begin Testing__

In [64]:
#Set up resource manager to handle pdf content. text, images, etc.
resource_manager = PDFResourceManager()

#Used to display text
fake_file_handle = io.StringIO()

#Set up converter
converter = TextConverter(resource_manager, fake_file_handle, laparams=LAParams())

#Set up page interpreter
page_interpreter = PDFPageInterpreter(resource_manager, converter)

with open('/Users/johnathonsmith/Downloads/293RDCV.PEND.CASE_1642601189564_-1416322429.pdf', 'rb') as fh:

    for page_number, page in enumerate(PDFPage.get_pages(fh,
                                  caching=True,
                                  maxpages = 14,
                                  check_extractable=True)):
        #Only read page 1 for now (index = 0)
        if page_number == 0:
            page_interpreter.process_page(page)

    text = fake_file_handle.getvalue()

# close open handles
converter.close()
fake_file_handle.close()

print(text)

                COUNTY OF MAVERICK 293RD DISTRICT COURT CIVIL PENDING CASES AS OF 01/19/2022

                                           RAN ON JANUARY 19, 2022 09:02am PAGE: 1

        CAUSE #             FILE DATE CAUSE OF ACTION           DOCKT DATE  DOCKET TYPE       ANS FILED  CR CASE #

PLAINTIFF NAME                PLAINTIFF ATTORNEY        DEFENDANT NAME                DEFENDANT ATTORNEY

        7815-OTH           02/22/1985 1                                                                  

        ATTORNEY GENERAL              MARIA LARSON              MENDEZ, CARLOS GILBERTO                            
        MENDEZ, OLGA LIDIA                                                                                         
        -------------------------------------------------------------------------------------------------------------
        90-11-10256-CV     11/08/1990 1                                                                  

        ATTORNEY GENERAL            

In [65]:
#Begin testing function
my_list = extract_pdf_data(text)

In [66]:
my_list

[{'County': 'Maverick',
  'Cause Number': '7815-OTH',
  'File Date': '02/22/1985',
  'Cause of Action': '1',
  'Docket Date': '',
  'Docket Type': '',
  'ANS File': '',
  'CR Number': '',
  'Plaintiff Name': ['ATTORNEY GENERAL', 'MENDEZ, OLGA LIDIA'],
  'Plaintiff Attorney': ['MARIA LARSON'],
  'Defendant Name': ['MENDEZ, CARLOS GILBERTO'],
  'Defendant Attorney': []},
 {'County': 'Maverick',
  'Cause Number': '90-11-10256-CV',
  'File Date': '11/08/1990',
  'Cause of Action': '1',
  'Docket Date': '',
  'Docket Type': '',
  'ANS File': '',
  'CR Number': '',
  'Plaintiff Name': ['ATTORNEY GENERAL', 'MARTINEZ, ROSALINDA'],
  'Plaintiff Attorney': ['ROLANDO MENCHACA'],
  'Defendant Name': ['MARTINEZ, HECTOR RENE'],
  'Defendant Attorney': []},
 {'County': 'Maverick',
  'Cause Number': '90-12-10326-CV',
  'File Date': '12/14/1990',
  'Cause of Action': 'INJURY OR DAMAGE OTHER T',
  'Docket Date': '',
  'Docket Type': '',
  'ANS File': '',
  'CR Number': '',
  'Plaintiff Name': ['DIAZ CAR

In [67]:
pd.DataFrame(my_list)

County    Cause Number   File Date           Cause of Action Docket Date  \
0  Maverick        7815-OTH  02/22/1985                         1               
1  Maverick  90-11-10256-CV  11/08/1990                         1               
2  Maverick  90-12-10326-CV  12/14/1990  INJURY OR DAMAGE OTHER T               
3  Maverick  91-06-10598-CV  10/22/1993  MOTION ENFORCEMENT, TO M  09/13/1994   
4  Maverick  95-01-12880-CV  01/05/1995                         1               
5  Maverick  96-05-13796-CV  05/08/1996                         1               
6  Maverick  96-12-14214-CV  12/20/1996         OTHER CIVIL CASES  02/18/1998   

        Docket Type ANS File CR Number  \
0                                        
1                                        
2                                        
3  MOTION TO ENFORC                      
4                                        
5                                        
6    WON JURY TRIAL                      

                               Plaintiff Name   Plaintiff Attorney  \
0      [ATTORNEY GENERAL, MENDEZ, OLGA LIDIA]       [MARIA LARSON]   
1     [ATTORNEY GENERAL, MARTINEZ, ROSALINDA]   [ROLANDO MENCHACA]   
2                 [DIAZ CARPENTER, ENRIQUETA]   [JAY T. KIMBROUGH]   
3  [ATTORNEY GENERAL, FUENTES, MARIA EUGENIA]   [ROLANDO MENCHACA]   
4   [ATTORNEY GENERAL, LARA, MARIA GUADALUPE]  [ENRIQUE FERNANDEZ]   
5      [ATTORNEY GENERAL, HERNANDEZ, OLGA E.]       [POLLY J BONE]   
6                        [THE STATE OF TEXAS]    [MELISSA DAYLONG]   

                   Defendant Name Defendant Attorney  
0       [MENDEZ, CARLOS GILBERTO]                 []  
1         [MARTINEZ, HECTOR RENE]                 []  
2              [BOWLES JR., RUDY]  [RICHARD TINSMAN]  
3          [ARROYO, JUAN ANTONIO]                 []  
4                  [LARA, HOMERO]                 []  
5            [HERNANDEZ, JUAN F.]                 []  
6  [APPROX. 440 POUNDS OF MARIHU]                 []

# Success!

Now I will fix the multiple pages issue.

In [68]:
#Load two pages
#Set up resource manager to handle pdf content. text, images, etc.
resource_manager = PDFResourceManager()

#Used to display text
fake_file_handle = io.StringIO()

#Set up converter
converter = TextConverter(resource_manager, fake_file_handle, laparams=LAParams())

#Set up page interpreter
page_interpreter = PDFPageInterpreter(resource_manager, converter)

with open('/Users/johnathonsmith/Downloads/293RDCV.PEND.CASE_1642601189564_-1416322429.pdf', 'rb') as fh:

    for page_number, page in enumerate(PDFPage.get_pages(fh,
                                  caching=True,
                                  maxpages = 2,
                                  check_extractable=True)):
        #Only read page 1 for now (index = 0)
        #if page_number == 0:
        page_interpreter.process_page(page)

    text = fake_file_handle.getvalue()

# close open handles
converter.close()
fake_file_handle.close()

print(text)

                COUNTY OF MAVERICK 293RD DISTRICT COURT CIVIL PENDING CASES AS OF 01/19/2022

                                           RAN ON JANUARY 19, 2022 09:02am PAGE: 1

        CAUSE #             FILE DATE CAUSE OF ACTION           DOCKT DATE  DOCKET TYPE       ANS FILED  CR CASE #

PLAINTIFF NAME                PLAINTIFF ATTORNEY        DEFENDANT NAME                DEFENDANT ATTORNEY

        7815-OTH           02/22/1985 1                                                                  

        ATTORNEY GENERAL              MARIA LARSON              MENDEZ, CARLOS GILBERTO                            
        MENDEZ, OLGA LIDIA                                                                                         
        -------------------------------------------------------------------------------------------------------------
        90-11-10256-CV     11/08/1990 1                                                                  

        ATTORNEY GENERAL            

In [69]:
#Try using regex to remove the headers on all pages
import re

In [70]:
re.sub(r"""COUNTY OF MAVERICK 293RD DISTRICT COURT CIVIL PENDING CASES AS OF 01/19/2022

                                           RAN ON JANUARY 19, 2022 09:02am PAGE: \d{1,3}

        CAUSE #             FILE DATE CAUSE OF ACTION           DOCKT DATE  DOCKET TYPE       ANS FILED  CR CASE #

PLAINTIFF NAME                PLAINTIFF ATTORNEY        DEFENDANT NAME                DEFENDANT ATTORNEY""", '', text)

'                \n\n        7815-OTH           02/22/1985 1                                                                  \n\n        ATTORNEY GENERAL              MARIA LARSON              MENDEZ, CARLOS GILBERTO                            \n        MENDEZ, OLGA LIDIA                                                                                         \n        -------------------------------------------------------------------------------------------------------------\n        90-11-10256-CV     11/08/1990 1                                                                  \n\n        ATTORNEY GENERAL              ROLANDO MENCHACA          MARTINEZ, HECTOR RENE                              \n        MARTINEZ, ROSALINDA                                                                                        \n        -------------------------------------------------------------------------------------------------------------\n        90-12-10326-CV     12/14/1990 INJURY OR DAMAGE 

In [71]:
#Test function on two pages
#First reload the two pages
#Load two pages
#Set up resource manager to handle pdf content. text, images, etc.
resource_manager = PDFResourceManager()

#Used to display text
fake_file_handle = io.StringIO()

#Set up converter
converter = TextConverter(resource_manager, fake_file_handle, laparams=LAParams())

#Set up page interpreter
page_interpreter = PDFPageInterpreter(resource_manager, converter)

with open('/Users/johnathonsmith/Downloads/293RDCV.PEND.CASE_1642601189564_-1416322429.pdf', 'rb') as fh:

    for page_number, page in enumerate(PDFPage.get_pages(fh,
                                  caching=True,
                                  maxpages = 3,
                                  check_extractable=True)):
        #Only read page 1 for now (index = 0)
        #if page_number == 0:
        page_interpreter.process_page(page)

    text = fake_file_handle.getvalue()

# close open handles
converter.close()
fake_file_handle.close()

print(text)

                COUNTY OF MAVERICK 293RD DISTRICT COURT CIVIL PENDING CASES AS OF 01/19/2022

                                           RAN ON JANUARY 19, 2022 09:02am PAGE: 1

        CAUSE #             FILE DATE CAUSE OF ACTION           DOCKT DATE  DOCKET TYPE       ANS FILED  CR CASE #

PLAINTIFF NAME                PLAINTIFF ATTORNEY        DEFENDANT NAME                DEFENDANT ATTORNEY

        7815-OTH           02/22/1985 1                                                                  

        ATTORNEY GENERAL              MARIA LARSON              MENDEZ, CARLOS GILBERTO                            
        MENDEZ, OLGA LIDIA                                                                                         
        -------------------------------------------------------------------------------------------------------------
        90-11-10256-CV     11/08/1990 1                                                                  

        ATTORNEY GENERAL            

In [72]:
#Now call the function
my_list = extract_pdf_data(text)

In [73]:
my_list

[{'County': 'Maverick',
  'Cause Number': '7815-OTH',
  'File Date': '02/22/1985',
  'Cause of Action': '1',
  'Docket Date': '',
  'Docket Type': '',
  'ANS File': '',
  'CR Number': '',
  'Plaintiff Name': ['ATTORNEY GENERAL', 'MENDEZ, OLGA LIDIA'],
  'Plaintiff Attorney': ['MARIA LARSON'],
  'Defendant Name': ['MENDEZ, CARLOS GILBERTO'],
  'Defendant Attorney': []},
 {'County': 'Maverick',
  'Cause Number': '90-11-10256-CV',
  'File Date': '11/08/1990',
  'Cause of Action': '1',
  'Docket Date': '',
  'Docket Type': '',
  'ANS File': '',
  'CR Number': '',
  'Plaintiff Name': ['ATTORNEY GENERAL', 'MARTINEZ, ROSALINDA'],
  'Plaintiff Attorney': ['ROLANDO MENCHACA'],
  'Defendant Name': ['MARTINEZ, HECTOR RENE'],
  'Defendant Attorney': []},
 {'County': 'Maverick',
  'Cause Number': '90-12-10326-CV',
  'File Date': '12/14/1990',
  'Cause of Action': 'INJURY OR DAMAGE OTHER T',
  'Docket Date': '',
  'Docket Type': '',
  'ANS File': '',
  'CR Number': '',
  'Plaintiff Name': ['DIAZ CAR

# Success!

Now try extracting the whole document and creating a dataframe.

In [74]:
df = build_cases_dataframe('/Users/johnathonsmith/Downloads/293RDCV.PEND.CASE_1642601189564_-1416322429.pdf')

Collected Data From 1179 Cases.


In [75]:
df

County     Cause Number   File Date           Cause of Action  \
0     Maverick         7815-OTH  02/22/1985                         1   
1     Maverick   90-11-10256-CV  11/08/1990                         1   
2     Maverick   90-12-10326-CV  12/14/1990  INJURY OR DAMAGE OTHER T   
3     Maverick   91-06-10598-CV  10/22/1993  MOTION ENFORCEMENT, TO M   
4     Maverick   95-01-12880-CV  01/05/1995                         1   
...        ...              ...         ...                       ...   
1174  Maverick  22-01-40738-MCV  01/14/2022        DIVORCE - CHILDREN   
1175  Maverick  22-01-40740-MCV  01/14/2022  DEBT/CONTRACT - DEBT COL   
1176  Maverick  22-01-40742-MCV  01/14/2022  INJURY OR DAMAGE INVOLVI   
1177  Maverick  22-01-40744-MCV  01/17/2022  DEBT/CONTRACT - DEBT COL   
1178  Maverick  22-01-40748-MCV  01/18/2022             JUDGMENT NISI   

     Docket Date       Docket Type ANS File CR Number  \
0                                                       
1                                                       
2                                                       
3     09/13/1994  MOTION TO ENFORC                      
4                                                       
...          ...               ...      ...       ...   
1174                                                    
1175                                                    
1176                                                    
1177                                                    
1178                                                    

                                  Plaintiff Name  \
0         [ATTORNEY GENERAL, MENDEZ, OLGA LIDIA]   
1        [ATTORNEY GENERAL, MARTINEZ, ROSALINDA]   
2                    [DIAZ CARPENTER, ENRIQUETA]   
3     [ATTORNEY GENERAL, FUENTES, MARIA EUGENIA]   
4      [ATTORNEY GENERAL, LARA, MARIA GUADALUPE]   
...                                          ...   
1174                           [ABOYTES CYNTHIA]   
1175                            [SYNCHRONY BANK]   
1176        [CASARES JR CARLOS, CASARES EDUARDO]   
1177               [CROWN ASSET MANAGEMENT, LLC]   
1178                             [SERNA ROBERTO]   

                      Plaintiff Attorney             Defendant Name  \
0                         [MARIA LARSON]  [MENDEZ, CARLOS GILBERTO]   
1                     [ROLANDO MENCHACA]    [MARTINEZ, HECTOR RENE]   
2                     [JAY T. KIMBROUGH]         [BOWLES JR., RUDY]   
3                     [ROLANDO MENCHACA]     [ARROYO, JUAN ANTONIO]   
4                    [ENRIQUE FERNANDEZ]             [LARA, HOMERO]   
...                                  ...                        ...   
1174                      [JAD P HARPER]           [ABOYTES ANDRES]   
1175                     [ZHANG JESSICA]             [CASAS EMILIO]   
1176  [VICTOR E. PERRY, VICTOR E. PERRY]            [FUENTES DAVID]   
1177              [BROGAN COLIN PATRICK]         [BALDERAS PAULA J]   
1178                                  []   [MEDIDORO ROILAN MARTIN]   

     Defendant Attorney  
0                    []  
1                    []  
2     [RICHARD TINSMAN]  
3                    []  
4                    []  
...                 ...  
1174                 []  
1175                 []  
1176                 []  
1177                 []  
1178                 []  

[1179 rows x 12 columns]

# Nearly a Success! 

It looks like my function failed to pick up 6 out of the 1,179 total cases. I need to figure that out next. After looking through the original PDF, I believe I have found the problem. Sometimes, when a case ends, and a new page begins, the PDF fails to generate the dashed line. I used the dashed line to separate the cases, so without it, some of them have been put together.

In [76]:
#Load an example of the above problem. I found it on pages 21 and 22 of the PDF
#Load two pages
#Set up resource manager to handle pdf content. text, images, etc.
resource_manager = PDFResourceManager()

#Used to display text
fake_file_handle = io.StringIO()

#Set up converter
converter = TextConverter(resource_manager, fake_file_handle, laparams=LAParams())

#Set up page interpreter
page_interpreter = PDFPageInterpreter(resource_manager, converter)

with open('/Users/johnathonsmith/Downloads/293RDCV.PEND.CASE_1642601189564_-1416322429.pdf', 'rb') as fh:

    for page_number, page in enumerate(PDFPage.get_pages(fh,
                                  caching=True,
                                  maxpages = 23,
                                  check_extractable=True)):
        #Only read page 21 and 22 for now (index = 20, 21)
        if page_number == 20 or page_number == 21:
            page_interpreter.process_page(page)

    text = fake_file_handle.getvalue()

# close open handles
converter.close()
fake_file_handle.close()

print(text)

                COUNTY OF MAVERICK 293RD DISTRICT COURT CIVIL PENDING CASES AS OF 01/19/2022

                                           RAN ON JANUARY 19, 2022 09:02am PAGE: 21

        CAUSE #             FILE DATE CAUSE OF ACTION           DOCKT DATE  DOCKET TYPE       ANS FILED  CR CASE #

PLAINTIFF NAME                PLAINTIFF ATTORNEY        DEFENDANT NAME                DEFENDANT ATTORNEY

        15-01-30572-MCV    12/16/2019 DIVORCE - NO CHILDREN     09/10/2021  STATUS HEARING               

        VALDEZ, ALEJANDRA SOFIA       PRISCILLA PUENTE-CHACON   ROCHA, FERNANDO ANTONIO       FRANCISCO MARTINEZ II
        -------------------------------------------------------------------------------------------------------------
        15-01-30610-MCV    01/21/2015 2                         11/09/2015  STATUS REPORT                

        ESPINOZA, MANUEL              WES HOLLAND               ALLSTATE VEHICLE AND PROPERT  ROGER D. HIGGINS     
        ESPINOZA, MARINA           

In [77]:
#Remove the headers using regex
text = re.sub(r"""COUNTY OF MAVERICK 293RD DISTRICT COURT CIVIL PENDING CASES AS OF .{10}

                                           RAN ON .{20,26} PAGE: \d{1,3}

        CAUSE #             FILE DATE CAUSE OF ACTION           DOCKT DATE  DOCKET TYPE       ANS FILED  CR CASE #

PLAINTIFF NAME                PLAINTIFF ATTORNEY        DEFENDANT NAME                DEFENDANT ATTORNEY""", '', text)

In [78]:
text

'                \n\n        15-01-30572-MCV    12/16/2019 DIVORCE - NO CHILDREN     09/10/2021  STATUS HEARING               \n\n        VALDEZ, ALEJANDRA SOFIA       PRISCILLA PUENTE-CHACON   ROCHA, FERNANDO ANTONIO       FRANCISCO MARTINEZ II\n        -------------------------------------------------------------------------------------------------------------\n        15-01-30610-MCV    01/21/2015 2                         11/09/2015  STATUS REPORT                \n\n        ESPINOZA, MANUEL              WES HOLLAND               ALLSTATE VEHICLE AND PROPERT  ROGER D. HIGGINS     \n        ESPINOZA, MARINA                                        STEWART JR., ALLAN                                 \n        -------------------------------------------------------------------------------------------------------------\n        15-01-30612-MCV    01/21/2015 DIVORCE - CHILDREN        01/07/2022  ENTRY OF ORDER               \n                           10/30/2019\n\n         GARZA, DIANA   

In [79]:
#Now split on the dashed lines
text = text.split('-------------------------------------------------------------------------------------------------------------')

In [80]:
#How many splits were generated?
len(text)

15

In [81]:
text[7]

'\n        15-02-30752-MCV    02/11/2015 OTHER CIVIL CASES         06/08/2015  ENTRY OF JUDGMEN             \n\n        MONSA, LTD                    BLAYNE FISHER             DOBBINS, STEVE                ELIZABETH SANDOVAL CA\n                                                                STATE FARM LLOYDS                                  \n\n\x0c                \n\n        15-02-30758-MCV    02/12/2015 2                         06/08/2015  PRE-TRIAL & DOCK             \n\n        MONSA, LTD                    BLAYNE FISHER             RASMUSSEN, CURTIS             ELIZABETH SANDOVAL CA\n                                                                STATE FARM LLOYDS                                  \n        '

In [82]:
def is_whitespace(x):
    if x.isspace() == True:
        return False
    else:
        return True

In [83]:
df = build_cases_dataframe('/Users/johnathonsmith/Downloads/293RDCV.PEND.CASE_1642601189564_-1416322429.pdf')

Collected Data From 1179 Cases.


# Getting there

Function still needs work

In [84]:
#Create a function to identify combined cases
def check_for_combined_case(string):
    """
    This function takes in a string from the case. Normally, this string would represent the plaintiff's name,
    but sometimes the cases can be combined in the pdf. So, this function will check if this string is actually
    a cause number. If the string contains the letters 'cv' or 'tx', it will return True.
    
    Parameter:
        - string: The string to be checked
        
    Returns:
        - Boolean: True or False
    """
    
    #If the string contains a hyphen or the letters 'cv', return True
    if string.count('CV') > 0 or string.count('TX') > 0:
        return True
    else:
        return False

# Testing Fix

I added a recursive function to extract case information, including cases that were combined into one. Testing below.

In [85]:
df = build_cases_dataframe('/Users/johnathonsmith/Downloads/293RDCV.PEND.CASE_1642601189564_-1416322429.pdf')

Collected Data From 1179 Cases.


In [87]:
df.head(20)

County       Cause Number       File Date           Cause of Action  \
0   Maverick           7815-OTH      02/22/1985                         1   
1   Maverick     90-11-10256-CV      11/08/1990                         1   
2   Maverick     90-12-10326-CV      12/14/1990  INJURY OR DAMAGE OTHER T   
3   Maverick     91-06-10598-CV      10/22/1993  MOTION ENFORCEMENT, TO M   
4   Maverick     95-01-12880-CV      01/05/1995                         1   
5   Maverick     96-05-13796-CV      05/08/1996                         1   
6   Maverick     96-12-14214-CV      12/20/1996         OTHER CIVIL CASES   
7   Maverick     97-06-14568-CV      06/16/1997         OTHER CIVIL CASES   
8   Maverick     97-10-14776-CV      10/10/1997  INJURY OR DAMAGE INVOLVI   
9   Maverick  97-12-14881-CVCLM  LM  12/15/1997         OTHER CIVIL CASES   
10  Maverick     98-03-15060-CV      03/19/1998  MOTION ENFORCEMENT, TO M   
11  Maverick     98-05-15216-CV      05/26/1998         OTHER CIVIL CASES   
12  Maverick     98-10-15536-CV      10/26/1998         OTHER CIVIL CASES   
13  Maverick     99-05-15966-CV      05/07/1999     DIVORCE - NO CHILDREN   
14  Maverick   95-07-13228-CV-B  B   05/10/1999  INJURY OR DAMAGE OTHER T   
15  Maverick   98-05-15224-CV-A  A   07/21/1999         OTHER CIVIL CASES   
16  Maverick     00-02-16392-CV      02/04/2000  (TITLE IV-D OAG USE ONLY   
17  Maverick    01-12-17744-MCV      12/06/2001         OTHER CIVIL CASES   
18  Maverick    02-06-18350-MCV      06/05/2002  (TITLE IV-D OAG USE ONLY   
19  Maverick    02-07-18488-MCV      07/17/2002                         1   

   Docket Date       Docket Type ANS File CR Number  \
0                                                     
1                                                     
2                                                     
3   09/13/1994  MOTION TO ENFORC                      
4                                                     
5                                                     
6   02/18/1998    WON JURY TRIAL                      
7   02/19/1998    WON JURY TRIAL                      
8   12/06/1999  TRIAL ON THE MER                      
9   10/21/2002        JURY TRIAL                      
10                                                    
11  01/10/2003     STATUS REPORT                      
12  05/11/2000  HEARING ON MERIT                      
13  07/29/2002         PRE-TRIAL                      
14                                                    
15                                                    
16                                                    
17  04/12/2002  TRIAL ON THE MER                      
18                                                    
19                                                    

                                       Plaintiff Name     Plaintiff Attorney  \
0              [ATTORNEY GENERAL, MENDEZ, OLGA LIDIA]         [MARIA LARSON]   
1             [ATTORNEY GENERAL, MARTINEZ, ROSALINDA]     [ROLANDO MENCHACA]   
2                         [DIAZ CARPENTER, ENRIQUETA]     [JAY T. KIMBROUGH]   
3          [ATTORNEY GENERAL, FUENTES, MARIA EUGENIA]     [ROLANDO MENCHACA]   
4           [ATTORNEY GENERAL, LARA, MARIA GUADALUPE]    [ENRIQUE FERNANDEZ]   
5              [ATTORNEY GENERAL, HERNANDEZ, OLGA E.]         [POLLY J BONE]   
6                                [THE STATE OF TEXAS]      [MELISSA DAYLONG]   
7                          [NUTMEG INSURANCE COMPANY]       [LOREN R. SMITH]   
8                        [ALLSTATE INSURANCE COMPANY]           [TIM HERRON]   
9   [GARZA, LEONEL, GARZA, MARTHA, HERNANDEZ, ALMA...  [GLORIA E. HERNANDEZ]   
10           [ATTORNEY GENERAL, FLORES, MARIA ISABEL]         [POLLY J BONE]   
11                              [PILLING, LISA DIONA]     [JERALD L. ABRAMS]   
12                 [GARCIA, MARTIN, GARCIA, SOILA M.]   [JOHN MICHAEL DOYLE]   
13                      [MARTINEZ, JUANITA GUADALUPE]                     []   
14                            [C

# Got It!

I believe the problem has been solved! Next step is to build a function for the criminal case PDFs. I'll work on that later.